packages/burner-wallet-connector/src/BurnerWalletConnector.ts

```diff
@@ -0,0 +1,262 @@
+  }
+
+  async connect(): Promise<boolean> {
+    if (!(await this.isConnected())) {
+      if (!this.burnerWalletProvider) {
+        throw Error('Burner Wallet Provider not found');
+      }
+
+      await this.setupBurnerWallet();
+
+      this.burnerWalletProvider = this.burnerWallet?.connect(
+        this.burnerWalletProvider,
+      ) as Provider;
+
+      this.storage.setItem(BURNER_WALLET_CONNECTED, 'true');
+    }
+
+    this.emit(this.events.connection, true);
+    this.emit(
+      this.events.currentAccount,
+      this.burnerWallet?.address.toAddress(),
+    );
+    this.emit(this.events.accounts, [this.burnerWallet?.address.toAddress()]);
+
+    return this.connected;
+  }
+
+  async accounts(): Promise<string[]> {
+    const account = this.burnerWallet?.address.toAddress();
+
+    if (!account) {
+      return [];
+    }
+
+    return [account as `fuel${string}`];
+  }
+
+  async disconnect(): Promise<boolean> {
+    if (await this.isConnected()) {
+      this.burnerWalletPrivateKey = null;
+      this.burnerWalletProvider = null;
+      this.burnerWallet = null;
+    }
+
+    this.storage.setItem(BURNER_WALLET_CONNECTED, 'false');
+
+    this.emit(this.events.connection, false);
+    this.emit(this.events.currentAccount, null);
+    this.emit(this.events.accounts, []);
+
+    return this.connected;
+  }
+
+  async signMessage(_address: string, _message: string): Promise<string> {
+    const signMessage = await this.burnerWallet?.signMessage(_message);
+
+    return signMessage as string;
+  }
+
+  async sendTransaction(
+    _address: string,
+    transaction: TransactionRequestLike,
+  ): Promise<string> {
+    const transactionRequest = await this.burnerWallet?.sendTransaction(
+      transaction,
+      { awaitExecution: true },
+    );
+
+    if (!transactionRequest) {
+      throw Error('Transaction request not found');
+    }
+
+    return transactionRequest?.id;
+  }
+
+  async currentAccount(): Promise<string | null> {
+    return this.burnerWallet?.address.toString() || null;
+  }
+
+  async addAssets(_assets: Asset[]): Promise<boolean> {
+    throw new Error('Method not implemented.');
+  }
+
+  async addAsset(_asset: Asset): Promise<boolean> {
+    throw new Error('Method not implemented.');
+  }
+
+  async assets(): Promise<Array<Asset>> {
+    return [];
+  }
+
+  async addNetwork(_networkUrl: string): Promise<boolean> {
+    throw new Error('Method not implemented.');
+  }
+
+  async selectNetwork(_network: Network): Promise<boolean> {
+    throw new Error('Method not implemented.');
+  }
+
+  async networks(): Promise<Network[]> {
+    return [await this.currentNetwork()];
+  }
+
+  async currentNetwork(): Promise<Network> {
+    if (!this.burnerWalletProvider) {
+      throw Error('Burner Wallet Provider not found');
+    }
+
+    const { chainId } = await this.burnerWalletProvider.getNetwork();
+
+    return {
+      chainId: Number(chainId),
+      url: this.burnerWalletProvider.url,
+    };
```

Comment:
This should return even if the wallet is not created. Because by spec this method does not require a authorization.